### PR TITLE
bugfix: properly forward error/warning count to and from wrapped scalac reporter

### PR DESCRIPTION
**Pull request by tgodzik**
Possibly fixes https://github.com/scalameta/scalameta/pull/3107

### CODE REVIEW

File: semanticdb/scalac/library/src/main/scala-2.12.17/scala/meta/internal/semanticdb/scalac/SemanticdbReporter.scala [hunk 1]
```diff
@@ -19,15 +19,20 @@ class SemanticdbReporter(underlying: Reporter)
...
 
-}
 
+  // overriding increment is enough so make sure that error/warning
+  // counts are the same as in underlying reporter
+   override def increment(severity: Severity): Unit = {
+    super.increment(severity)
+    underlying.increment(severity)
+  }
+}
```
Comment (bmarker) [inline semanticdb/scalac/library/src/main/scala-2.12.17/scala/meta/internal/semanticdb/scalac/SemanticdbReporter.scala:35]: This does not appear to be sufficient. This did not fix my issues with Silencer while the full Scala 2.13 version of `SemanticdbReporter.scala` works.

File: semanticdb/scalac/library/src/main/scala-2.12.17/scala/meta/internal/semanticdb/scalac/SemanticdbReporter.scala [hunk 2]
```diff
@@ -19,15 +19,20 @@ class SemanticdbReporter(underlying: Reporter)
...
   override def hasWarnings: Boolean = underlying.hasWarnings
 
-}
 
+  // overriding increment is enough so make sure that error/warning
+  // counts are the same as in underlying reporter
+   override def increment(severity: Severity): Unit = {
+    super.increment(severity)
+    underlying.increment(severity)
+  }
```
Comment (bmarker) [inline semanticdb/scalac/library/src/main/scala-2.12.17/scala/meta/internal/semanticdb/scalac/SemanticdbReporter.scala:34]: This does not appear to be sufficient. The compile succeeds without any WartRemover errors.

**Comment by bmarker** [comment]
PR #3108 (with the path removed):
```
[info] compiling 130 Scala sources to ***/target/scala-2.12/classes ...
[error] ***.scala:21:43: [wartremover:IterableOps] head is disabled - use headOption instead
[error]         case (errors, _)   => Left(errors.head.left.get)
[error]                                           ^
[error] one error found
[error] (Compile / compileIncremental) Compilation failed
[error] Total time: 28 s, completed Apr 8, 2023 8:26:29 PM
```

PR #3107 (with the path removed):
```
[info] compiling 130 Scala sources to ***/target/scala-2.12/classes ...
[info] Running scalafix on 129 Scala sources
[info] ℹ On latest GA version of Talon: 2.8.3
[success] Total time: 41 s, completed Apr 8, 2023 8:30:43 PM
```

**Comment by kitbellew** [comment]
@tgodzik is this still relevant?

**Comment by tgodzik** [comment]
No anymore! Thanks for reminding me!

**Pull request closed without merging**